app/proxy.js

```diff
@@ -30,28 +52,47 @@ function tryAgain(context) {
  * that can be replaced for various authentication formats and protocols.
  */
 var dummy = function(context, settings, callback) {
-  callback(context.login==settings.login && context.pw==settings.password);
+  callback(context.login && context.login==settings.login && context.pw==settings.password);
+};
+
+function authenticateIfPresent(context, callback) {
+  var authenticationPresent=context.login ? true : false;
+  authenticate(context,callback,!authenticationPresent);
 };
 
 function authenticate(context, callback, shouldNotCatch) {
-  if (context.requestIn.headers.authorization) {
     var authenticators = configuration.sites[context.conf].authentication;
     async.detectSeries(authenticators, function(authenticator, callback) {
       if (authenticator.dn) {
         ldap(context, authenticator, callback);
+      } else if (authenticator.hasOwnProperty("cookieName")) {
```

Comment:
I think this block is never run, since it looks for `cookieName` at `configuration.sites[i].authentication[j].cookieName`.
The only documented `cookieName` is at `configuration.sites[i].sessionHandler`.

Is this block an undocumented feature or a cruft?